test/jdk/java/awt/dnd/RejectDragTest.java

```diff
@@ -0,0 +1,172 @@
+    private static volatile Point startPoint;
+    private static volatile Point endPoint;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+
+            EventQueue.invokeAndWait(RejectDragTest::createAndShowUI);
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            EventQueue.invokeAndWait(RejectDragTest::addDnDListeners);
+            robot.waitForIdle();
+
+            testDnD();
+            robot.waitForIdle();
+            robot.delay(200);
+        } finally {
+            EventQueue.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    private static void addDnDListeners() {
+        final DragSourceListener dragSourceListener = new DragSourceAdapter() {
+            private boolean first = true;
+
+            public void dragEnter(DragSourceDragEvent dsde) {
+                first = false;
+                dragEnterCalled = true;
+            }
+
+            public void dragExit(DragSourceEvent dse) {
+                if (first) {
+                    dragExitCalledAtFirst = true;
+                    first = false;
+                }
+            }
+
+            public void dragDropEnd(DragSourceDropEvent dsde) {
+                first = false;
+            }
+
+            public void dragOver(DragSourceDragEvent dsde) {
+                first = false;
+                dragOverCalled = true;
+            }
+
+            public void dropActionChanged(DragSourceDragEvent dsde) {
+                first = false;
+            }
+        };
+
+        DragGestureListener dragGestureListener =
+                dge -> dge.startDrag(null, new StringSelection("OKAY"),
+                                     dragSourceListener);
+        new DragSource().createDefaultDragGestureRecognizer(frame,
+                                                            DnDConstants.ACTION_COPY,
+                                                            dragGestureListener);
+
+        DropTargetAdapter dropTargetListener = new DropTargetAdapter() {
+            public void dragEnter(DropTargetDragEvent dtde) {
+               dtde.rejectDrag();
+            }
+
+            public void drop(DropTargetDropEvent dtde) {
+               dtde.rejectDrop();
+            }
+        };
+
+        new DropTarget(frame, dropTargetListener);
+    }
+
+    private static void createAndShowUI() {
+        frame = new Frame("RejectDragTest");
+        frame.setSize(200, 200);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static void testDnD() throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            startPoint = frame.getLocationOnScreen();
+            endPoint = new Point(startPoint);
+            startPoint.translate(50, 50);
+            endPoint.translate(150, 150);
```

Comment:
Let's make it correct, so that the object fields aren't modified after assigning the reference to the volatile field:
```suggestion
            Point start = frame.getLocationOnScreen();
            start.translate(50, 50);
            startPoint = start;
            
            Point end = new Point(start);
            end.translate(150, 150);
            endPoint = end;
```

Reply:
Updated